transport/server.cc

```diff
@@ -212,6 +218,101 @@ cql_server::cql_server(distributed<cql3::query_processor>& qp, auth::service& au
                             seastar::format("Holds the amount of available memory for admitting new requests (max is {}B)."
                                             "Zero value indicates that our bottleneck is memory and more specifically - the memory quota allocated for the \"CQL transport\" component.", _max_request_size))),
 
+        sm::make_derive("cql_errors_total", sm::description("Counts the total number of returned CQL errors."),
```

Comment:
It's not going to work this way
1. always register the metrics:
a.  you will not get a second chance to do it
b. you can not register the same metrics twice (two metrics with no label)
c. it's better to start a metrics early

Reply:
Alright, I understand when I return the empty label vector it's gonna create problems. And that I should initialize all entries early.
But is there a metric that is not registered? I don't get what you're getting at with 1 and a.

Reply:
in the original code, if a metric was not used yet (i.e. it will be used in the future) you will never register it.
The metrics registration code is called once, if it was not register then, it will never ber register.

I hope at least b,c, convinced you :)

Reply:
This just makes me more confused. What do you mean by "use" the metric? How can I use it before registering it?
I feel like I'm misunderstanding the terms here - isn't what I'm doing here "registering"? I thought when I call `add_group` and pass a metric created with `sm::make_derive` for example, it's what leads to registration of said metric.

Reply:
yes, the registration part (this code) is called once in this case during the contractor.
if you condition your metrics registration, i.g. you only register a metric if it was used, it will never be registered.

let me try to explain it differently
you have a map of errors and your map is empty and an entry is only added on the first time it is used (on the first error) you have a problem with the registration.
If you condition a registration on the existing of an entry, any entry that will not be available during the registration will never be registered even when it will be added at a future time.

Reply:
ok, that I understand
but I fail to see how this is conditioned on anything (apart from the situation where I return a separate label instance vector, but you mentioned that in a different point)

Reply:
Because in case of the error counting map, we attach the lambda to the metric and only that lambda checks the entry's existence. Is this where I'm mistaken?

Reply:
I gave you three reasons why not to condition your metrics registration on an entry existence, I think we've passed that and we agree you're not going to do it